### DISCUSSION TITLE
transmission-qt cannot be built with qt6 on macOS without ninja and raising the target to 10.15

### DISCUSSION THREAD

**Discussion by GaryElshaw** [transmission/transmission#4474] [General]
### What is the issue?

It's great that transmission-qt can be built on macOS with qt6 #4405, but _better_ would be for this process to be possible with make and not having to raise the target.

### Which application of Transmission?

Qt app on macOS.

### Which version of Transmission?

Beta 2

**Comment by mikedld** [comment]
I already noted in #4405 that the necessity to raise deployment target to 10.15 has nothing to do with Transmission. It's certainly possible to build Qt in a way that wouldn't require 10.15, in turn not transferring that requirement to Transmission. Building Qt on Mac though is not our job, and I doubt Homebrew people will be willing to accomodate such a request either because their project has different goals.

I also tried using other generators (Makefiles, Xcode) and they all generate working build system for both Qt 5 and 6, using the same steps described in #4405. Switching between Qt 5 and 6 within the same build directory is not supported, so you have to make sure to do a clean reconfiguration (which is different from building the `clean` target allowing for clean rebuild) if you want to switch. Please try configuring against Qt 6 with a clean build directory and your favorite generator and report back (maybe just re-open #4405 since it's still the same issue).

P.S. This is a combined issue and we don't like those very much, as was mentioned before by @ckerr on a few occasions. I'm converting it to a discussion until there's something actionable.

**Comment by GaryElshaw** [comment]
I'm trying to understand whether the issue that needs resolving to build with make is with qt@6, Transmission, or what is i think more likely, having both 5 and 6 installed.

Yep, i agree, converting to discussion is the best move.

**Comment by GaryElshaw** [comment]
I thought i'd try the cmake gui client (3.25.1) to see if any qt5 elements might have been pulled in to the mix. After adjusting the target, make built successfully. 

![SCR-20221227-kam](https://user-images.githubusercontent.com/69029666/209597654-c2ae0ac5-89d4-46ef-be4e-2c0d6b9fd534.png)
![SCR-20221227-kb1](https://user-images.githubusercontent.com/69029666/209597668-2cca74ba-abf6-45d9-b0cc-a84e598b88c0.png)
![SCR-20221227-kbd](https://user-images.githubusercontent.com/69029666/209597682-ff5e3304-d501-4d12-be82-2aab0677de16.png)

**Reply by GaryElshaw** [reply]
This also worked. Changing line 325 of the cmakelists file to `set(Qt_NAMES Qt6)`
